vars/checkChangeInDevelopment.groovy

```diff
@@ -32,20 +33,24 @@ import com.sap.piper.cm.ChangeManagementException
     'git_format'
   ]
 
+@Field Set generalConfigurationKeys = stepConfigurationKeys
+
 def call(parameters = [:]) {
 
     handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
 
-        prepareDefaultValues script: this
+        def script = parameters.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
 
         GitUtils gitUtils = parameters?.gitUtils ?: new GitUtils()
 
-        ChangeManagement cm = parameters?.cmUtils ?: new ChangeManagement(parameters.script, gitUtils)
-
-        Map configuration = ConfigurationMerger.merge(parameters.script, STEP_NAME,
-                                                      parameters, parameterKeys,
-                                                      stepConfigurationKeys)
+        ChangeManagement cm = parameters?.cmUtils ?: new ChangeManagement(script, gitUtils)
 
+        Map configuration = ConfigurationHelper
+                            .loadStepDefaults(this)
+                            .mixinGeneralConfig(script.commonPipelineEnvironment, generalConfigurationKeys)
+                            .mixinStepConfig(script.commonPipelineEnvironment, stepConfigurationKeys)
```

Comment:
we should also consider `stage` configuration

Reply:
So far I don't see any relation to a certain stage. This is why I omitted it.
In case we use the step(s) in different stages the values, e.g. endpoint will be the same all the time, regardless which stage. Hence I expect some properties in the general section ...

Reply:
@CCFenner : so far I do not see any stage context for the change mangement related steps.
Instead we have a - lets name it - landscape context which is not implemented for now.

Is it from your point of view a mistake not to define/use a stage context?

Reply:
The user may expect to be able to do stage specific configuration. We should be consistent here and consider general/stage/steps always in all steps.

Reply:
OK, this I understand. Since I do not want to make assumptions about the stage names. Is it possible to load configuration for the current stage within the step is called, regardless which name the current stage has?

Reply:
... found it: ```env.STAGE_NAME```

Reply:
Have a look here: https://github.com/SAP/jenkins-library/blob/master/vars/snykExecute.groovy#L30

We also gave the ability to override the stage name.